src/server/scripts/Spells/spell_item.cpp

```diff
@@ -3221,6 +3221,124 @@ class spell_item_taunt_flag_targeting : public SpellScriptLoader
+{
+    public:
+        spell_item_mind_control_cap() : SpellScriptLoader("spell_item_mind_control_cap") { }
+
+        class spell_item_mind_control_cap_SpellScript : public SpellScript
+        {
+            PrepareSpellScript(spell_item_mind_control_cap_SpellScript);
+
+            bool Load() override
+            {
+                if (!GetCastItem())
+                    return false;
+                return true;
+            }
+
+            bool Validate(SpellInfo const* /*spell*/) override
+            {
+                if (!sSpellMgr->GetSpellInfo(SPELL_GNOMISH_MIND_CONTROL_CAP) || !sSpellMgr->GetSpellInfo(SPELL_DULLARD))
+                    return false;
+                return true;
+            }
+
+            void HandleDummy(SpellEffIndex /* effIndex */)
+            {
+                Unit* caster = GetCaster();
+                if (Unit* target = GetHitUnit())
+                {
+                    if (roll_chance_i(ROLL_CHANCE_NO_BACKFIRE))
+                        caster->CastSpell(target, roll_chance_i(ROLL_CHANCE_DULLARD) ? SPELL_DULLARD : SPELL_GNOMISH_MIND_CONTROL_CAP, true, GetCastItem());
+                    else
+                        target->CastSpell(caster, SPELL_GNOMISH_MIND_CONTROL_CAP, true); // backfire - 5% chance
+                }
+            }
+
+            void Register() override
+            {
+                OnEffectHitTarget += SpellEffectFn(spell_item_mind_control_cap_SpellScript::HandleDummy, EFFECT_0, SPELL_EFFECT_DUMMY);
+            }
+        };
+
+        SpellScript* GetSpellScript() const override
+        {
+            return new spell_item_mind_control_cap_SpellScript();
+        }
+};
+
+// 8344 - Universal Remote (Gnomish Universal Remote)
+enum UniversalRemote
+{
+    SPELL_CONTROL_MACHINE       = 8345,
+    SPELL_MOBILITY_MALFUNCTION  = 8346,
+    SPELL_TARGET_LOCK           = 8347
+};
+
+static std::unordered_map<uint32 /*spellId*/, double /*probability*/> const UniversalRemoteActions =
+{
+    { SPELL_CONTROL_MACHINE,      60.0 },
+    { SPELL_MOBILITY_MALFUNCTION, 25.0 },
+    { SPELL_TARGET_LOCK,          15.0 }
+};
+
+class spell_item_universal_remote : public SpellScriptLoader
+{
+    public:
+        spell_item_universal_remote() : SpellScriptLoader("spell_item_universal_remote") { }
+
+        class spell_item_universal_remote_SpellScript : public SpellScript
+        {
+            PrepareSpellScript(spell_item_universal_remote_SpellScript);
+
+            bool Load() override
+            {
+                if (!GetCastItem())
+                    return false;
+                return true;
+            }
+
+            bool Validate(SpellInfo const* /*spellInfo*/) override
+            {
+                for (auto const& action : UniversalRemoteActions)
+                    if (!sSpellMgr->GetSpellInfo(action.first))
+                        return false;
+                return true;
+            }
+
+            void HandleDummy(SpellEffIndex /*effIndex*/)
```

Comment:
OK, so that means I can remove
```cpp
static std::unordered_map<uint32 /*spellId*/, double /*probability*/> const UniversalRemoteActions =
{
    { SPELL_CONTROL_MACHINE,      60.0 },
    { SPELL_MOBILITY_MALFUNCTION, 25.0 },
    { SPELL_TARGET_LOCK,          15.0 }
};
```
and replace that with enum instead. I will take a look at it as soon as I have done some other work.
```cpp
enum UniversalRemote
{
    CHANCE_TARGET_LOCK          = 15,
    CHANCE_MOBILITY_MALFUNCTION = 25,

    SPELL_CONTROL_MACHINE       = 8345,
    SPELL_MOBILITY_MALFUNCTION  = 8346,
    SPELL_TARGET_LOCK           = 8347
};
```
```cpp
if (Unit* target = GetHitUnit())
{
    uint8 chance = urand(0, 99);
    if (chance <= CHANCE_TARGET_LOCK)
        GetCaster()->CastSpell(target, SPELL_TARGET_LOCK, true, GetCastItem()); 
    else if (chance <= CHANCE_MOBILITY_MALFUNCTION)
        GetCaster()->CastSpell(target, SPELL_MOBILITY_MALFUNCTION, true, GetCastItem()); 
    else
        GetCaster()->CastSpell(target, SPELL_CONTROL_MACHINE, true, GetCastItem());
}
```
(is it possible to use `<=` meaning "less than or equal to" ?)